Philips.CodeAnalysis.Test/Maintainability/Maintainability/ReduceCognitiveLoadAnalyzerTest.cs

```diff
@@ -0,0 +1,469 @@
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1!=1) {}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 3 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadIfReturn()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1==1) {return;}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 3 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadIfLogicalOr()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1==1 || 2==2) {}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 3 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadIfLogicalAnd()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1==1 && 2==2) {}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 3 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadBreak()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		while (1<2) { break; }
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 3 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadBang()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (!(1==1)) {}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 3 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadNested1()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1==1)
+		{
+			// Nested block statement counts double
+			if (2 == 2) {}
+		}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 4 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadNested2()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1==1)
+		{
+			// Nested block statement counts double
+			if (2 == 2) {}
+			// Nested block statement counts double
+			if (3 == 3) {}
+		}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 6 ")));
+		}
+
+		[TestMethod]
+		public void CognitiveLoadNested3()
+		{
+			const string template = @"
+class Foo
+{
+	private void Test()
+	{
+		if (1==1)
+		{
+			// Nested block statement counts double
+			if (2 == 2)
+			{
+				// Nested block statement counts more than triple - it's exponential
+				if (3 == 3) {}
+			}
+		}
+	}
+}
+";
+			VerifyCSharpDiagnostic(template, DiagnosticResultHelper.Create(DiagnosticIds.ReduceCognitiveLoad, new Regex("Load of 8 ")));
```

Comment:
Consider making a method that takes just the expected count as parameter.

Reply:
done